Sources/Datadog/URLSessionAutoInstrumentation/Interception/URLSessionInterceptor.swift

```diff
@@ -6,18 +6,29 @@
 
 import Foundation
 
+/// An interface for handling `URLSession` interceptions start and completion.
+internal protocol URLSessionInterceptionHandler {
+    /// Notifies the `URLSessionTask` interception start.
+    func notify_taskInterceptionStarted(interception: TaskInterception)
+    /// Notifies the `URLSessionTask` interception completion.
+    func notify_taskInterceptionCompleted(interception: TaskInterception)
+}
+
+/// An interface for processing `URLSession` task interceptions.
 internal protocol URLSessionInterceptorType: class {
     func modify(request: URLRequest, session: URLSession?) -> URLRequest
     func taskCreated(task: URLSessionTask, session: URLSession?)
     func taskMetricsCollected(task: URLSessionTask, metrics: URLSessionTaskMetrics)
     func taskReceivedData(task: URLSessionTask, data: Data)
     func taskCompleted(task: URLSessionTask, error: Error?)
+
+    var handler: URLSessionInterceptionHandler { get }
```

Comment:
💡 This adds a bit noise to the `URLSessionInterceptorType` interface and could be additionally mitigated by decoupling the `handler` creation to upstream (to `URLSessionAutoInstrumentation` class). This however seems over engineering if done in the scope of this PR - we may want to refactor this on a stronger reason with more constraints for changing the abstraction.